src/Microsoft.Health.SqlServer/Features/Schema/Manager/BaseSchemaRunner.cs

```diff
@@ -80,9 +80,17 @@ await Policy.Handle<SchemaManagerException>()
 
         private async Task InstanceSchemaRecordCreatedAsync(CancellationToken cancellationToken)
         {
-            if (!await _schemaManagerDataStore.InstanceSchemaRecordExistsAsync(cancellationToken))
+            try
             {
-                throw new SchemaManagerException(Resources.InstanceSchemaRecordErrorMessage);
+                if (!await _schemaManagerDataStore.InstanceSchemaRecordExistsAsync(cancellationToken))
+                {
+                    throw new SchemaManagerException(Resources.InstanceSchemaRecordErrorMessage);
+                }
+            }
+            catch (SqlException e) when (e.Message.Contains("Invalid object name", StringComparison.OrdinalIgnoreCase))
```

Comment:
I am wondering when would this happen? Since before checking if record exists or not, we ensure that the base schema exists. So if it fails to create table as part of base schema execution, it should throw exception in _baseSchemaRunner.EnsureBaseSchemaExistsAsync.

Reply:
While that is true, EnsureInstanceSchemaRecordExistsAsync is a public method, so it's possible that the caller didn't initialize the base schema yet. Though I should make it have a different error message so it's clearer that the issue is the base schema wasn't initialized yet.